server/container_create_linux.go

```diff
@@ -232,16 +232,17 @@ func (s *Server) createContainerPlatform(container *oci.Container, infraContaine
 		}
 
 		runDirPath := filepath.Join(intermediateMountPoint, "rundir")
-		err = os.MkdirAll(runDirPath, 0700)
+		err = idtools.MkdirAllAs(runDirPath, 0700, rootUID, rootGID)
 		if err != nil {
 			return
 		}
-
-		err = unix.Mount(container.BundlePath(), runDirPath, "none", unix.MS_BIND, "suid")
+		err = chownAllFilesAt(container.BundlePath(), rootUID, rootGID)
```

Comment:
could we use `idtools.MkdirAllAndChown` instead of `MkdirAllAs` + `chownAllFilesAt`?

Reply:
@giuseppe mkdirAllAs and chownAllFilesAt are operating on different paths. `chownAllFilesAt` is operating on pre-created path and not the one created by just above mkdirAllAs. I am not sure how MkdirAllAndChown will be sufficient.

Reply:
`MkdirAllAndChown` should do the chown also for an existing path (it might not recurse the existing path though, so if there are other files/dirs below then they won't be updated).

can the existing path already have other files below it?

Reply:
anyway, not blocking comment, we can re-visit it later if necessary

Reply:
yes, there are files under container.BundlePath() that are also to be chowned.